src/utils.js

```diff
@@ -58,6 +58,57 @@ export function getRootExpression(node) {
 }
 
 /*
+ * Returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  // Finds variable reference in current scope.
+  for (let variable of variables) {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+      break;
+    }
+  }
+
+  if (scopeVar && scopeVar.defs && scopeVar.defs[0] &&
```

Comment:
This bit is a bit complex and required a bit of work due to the structure that eslint gives us AST nodes. I tried to comment it pretty well to make everything clear but let me know if you have any questions here.